doc/sphinx/practical-tools/utilities.rst

```diff
@@ -190,6 +190,85 @@ since it results in a “packed” plugin: All auxiliary modules (as
 (``Qux_plugin``). This reduces the chances of begin unable to load two
 distinct plugins because of a clash in their auxiliary module names.
 
+.. _arbitrary_arguments:
+
+Passing arbitrary arguments to coqc
++++++++++++++++++++++++++++++++++++
+Using either ``-arg "-w all"`` or ``-arg -w -arg all`` its possible to pass the
+option ``-w all`` to ``coqc``. For options like ``-set 'Default Goal
+Selector=!'`` quotes are necessary: ``-arg "-set 'Default Goal Selector=!'"``.
+It is impossible to pass a double quote using ``-arg``, only single quotes work
+in this situation.
+
+.. _comments:
+
+Comments
+++++++++
+The syntax for comments in a ``_CoqProject`` file is a bit unusual. The
+character ``#`` starts a comment until the end of the line only if it is
+outside a quoted string, and preceded by whitespace or a closing double quote.
+This means ``Filename#.v`` and ``"Filename With # Spaces.v"`` won’t cause
+comments. The ``#`` is in these cases treated like an ordinary character.
+
+.. _grammar_of_coqproject:
+
+The grammar of _CoqProject
+++++++++++++++++++++++++++
+The parser of coq_makefile uses the following grammar:
+
+  .. insertprodn CoqProject unquoted_string
+
+  .. prodn::
+     CoqProject ::= {* {| @blank | @comment | @quoted_string | @unquoted_string } }
+     blank ::= {| space | horizontal_tab | newline }
+     comment ::= # {* comment_byte } newline
+     quoted_string ::= " {* quoted_string_byte } "
+     unquoted_string ::= string_start_byte {* unquoted_string_byte }
+
+Where the following definitions apply:
+
+* :n:`space`, :n:`horizontal_tab` and :n:`newline` stand for the corresponding ASCII values.
+* :n:`comment_byte` is the set of all bytes except :n:`newline`.
+* :n:`quoted_string_byte` is the set of all bytes except ``"``.
+* :n:`string_start_byte` is the set of all bytes except those that match :n:`@blank`, or are ``"`` or ``#``.
+* :n:`unquoted_string_byte` is the set of all bytes except those that match :n:`@blank`.
+
+The parser produces a list of strings in the same order as they were
+encountered in ``_CoqProject``. All :n:`@blank` and :n:`@comment` tokens
+get ignored and the double quotes of :n:`@quoted_string` tokens are removed as
+well.
+
+The list of strings is then parsed using the following grammar:
+
+  .. insertprodn options option3
+
+  .. prodn::
+     options ::= {* @option }
+     option ::= {| @option0 | @option1 string | @option2 string string | @option3 string string string | string = string | -arg string | string }
+     option0 ::= {| -impredicative-set | -no-install | -custom | -no-opt | -byte | -full | -opt }
+     option1 ::= {| -install | -native-compiler | -f | -o }
+     option2 ::= {| -Q | -I | -R }
+     option3 ::= {| -extra | -extra-phony }
+
+The grammar given above might accept a given list of strings in two different
+ways, precisely if an equals-sign is involved. In this case the implementation
+chooses the leftmost interpretation.
```

Comment:
Please be more precise (and simpler, if possible).  Sounds like the equals sign is the only special case.

Reply:
Jup, will do.

Reply:
The equals sign belongs to deprecated behaviour, and (probably) shouldn’t be used anymore. It can still be used to add variable definitions to CoqMakefile.local. The line `FOO = BAR` in _CoqProject produces `FOO = BAR` in CoqMakefile.local, where BAR may be an arbitrary token and FOO must not be a recognized option (except `-arg` which is allowed).

Reply:
I changed the code, so `-arg` and `_ = _` interact differently, giving `_ = _` less precedence.